test/unit/datetime.c

```diff
@@ -0,0 +1,261 @@
+	S("2012-12-24 14:30:00-0100"),
+	S("2012-12-24 15:30:00.123456Z"),
+	S("2012-12-24 15:30:00.123456z"),
+	S("2012-12-24 15:30:00.123456"),
+	S("2012-12-24 16:30:00.123456+01:00"),
+	S("2012-12-24 16:30:00.123456+01"),
+	S("2012-12-24 14:30:00.123456-01:00"),
+	S("2012-12-24 14:30:00.123456-01"),
+	S("2012-12-24t15:30Z"),
+	S("2012-12-24t15:30z"),
+	S("2012-12-24t15:30"),
+	S("2012-12-24t16:30+01:00"),
+	S("2012-12-24t16:30+0100"),
+	S("2012-12-24t14:30-01:00"),
+	S("2012-12-24t14:30-0100"),
+	S("2012-12-24t15:30:00Z"),
+	S("2012-12-24t15:30:00z"),
+	S("2012-12-24t15:30:00"),
+	S("2012-12-24t16:30:00+01:00"),
+	S("2012-12-24t16:30:00+0100"),
+	S("2012-12-24t14:30:00-01:00"),
+	S("2012-12-24t14:30:00-0100"),
+	S("2012-12-24t15:30:00.123456Z"),
+	S("2012-12-24t15:30:00.123456z"),
+	S("2012-12-24t16:30:00.123456+01:00"),
+	S("2012-12-24t14:30:00.123456-01:00"),
+	S("2012-12-24 16:30 +01:00"),
+	S("2012-12-24 14:30 -01:00"),
+	S("2012-12-24 15:30 UTC"),
+	S("2012-12-24 16:30 UTC+1"),
+	S("2012-12-24 16:30 UTC+01"),
+	S("2012-12-24 16:30 UTC+0100"),
+	S("2012-12-24 16:30 UTC+01:00"),
+	S("2012-12-24 14:30 UTC-1"),
+	S("2012-12-24 14:30 UTC-01"),
+	S("2012-12-24 14:30 UTC-01:00"),
+	S("2012-12-24 14:30 UTC-0100"),
+	S("2012-12-24 15:30 GMT"),
+	S("2012-12-24 16:30 GMT+1"),
+	S("2012-12-24 16:30 GMT+01"),
+	S("2012-12-24 16:30 GMT+0100"),
+	S("2012-12-24 16:30 GMT+01:00"),
+	S("2012-12-24 14:30 GMT-1"),
+	S("2012-12-24 14:30 GMT-01"),
+	S("2012-12-24 14:30 GMT-01:00"),
+	S("2012-12-24 14:30 GMT-0100"),
+	S("2012-12-24 14:30 -01:00"),
+	S("2012-12-24 16:30:00 +01:00"),
+	S("2012-12-24 14:30:00 -01:00"),
+	S("2012-12-24 16:30:00.123456 +01:00"),
+	S("2012-12-24 14:30:00.123456 -01:00"),
+	S("2012-12-24 15:30:00.123456 -00:00"),
+	S("20121224T1630+01:00"),
+	S("2012-12-24T1630+01:00"),
+	S("20121224T16:30+01"),
+	S("20121224T16:30 +01"),
+};
+#undef S
+
+static int
+parse_datetime(const char *str, size_t len, int64_t *secs_p,
+	       int32_t *nanosecs_p, int32_t *offset_p)
+{
+	size_t n;
+	dt_t dt;
+	char c;
+	int sec_of_day = 0, nanosecond = 0, offset = 0;
+
+	n = dt_parse_iso_date(str, len, &dt);
+	if (!n)
+		return 1;
+	if (n == len)
+		goto exit;
+
+	c = str[n++];
+	if (!(c == 'T' || c == 't' || c == ' '))
+		return 1;
+
+	str += n;
+	len -= n;
+
+	n = dt_parse_iso_time(str, len, &sec_of_day, &nanosecond);
+	if (!n)
+		return 1;
+	if (n == len)
+		goto exit;
+
+	if (str[n] == ' ')
+		n++;
+
+	str += n;
+	len -= n;
+
+	n = dt_parse_iso_zone_lenient(str, len, &offset);
+	if (!n || n != len)
+		return 1;
+
+exit:
+	*secs_p = ((int64_t)dt_rdn(dt) - DT_EPOCH_1970_OFFSET) * SECS_PER_DAY +
+		  sec_of_day - offset * 60;
+	*nanosecs_p = nanosecond;
+	*offset_p = offset;
+
+	return 0;
+}
+
+static int
+local_rd(const struct datetime *dt)
+{
+	return (int)((int64_t)dt->epoch / SECS_PER_DAY) + DT_EPOCH_1970_OFFSET;
+}
+
+static int
+local_dt(const struct datetime *dt)
+{
+	return dt_from_rdn(local_rd(dt));
+}
+
+struct tm *
+datetime_to_tm(struct datetime *dt)
```

Comment:
The fact that you copied 3 functions from the .c datetime file looks like a problem. Most likely with the naming - you couldn't expose obfuscated names like `local_rd` and `local_dt` to the .h file, so you just copied them. I propose to give them proper names and expose in datetime.h/.c. Use `datetime_` prefix, stop using `dt` suffix whose meaning I still don't know but usually it looks like 'days since Christ' or 'Rate Die date' or something.

Because of the copy-paste you didn't add `static` to `datetime_to_tm()`, its argument is not constant, and you still use static tm inside.

Reply:
This copy-paste was from times where I could not refer to libcore.a as it was created with later commit. Now they are part of a same commit, and we do have such dependency already, so I could reuse core code further. Going to remove some copy-paste soon.

Reply:
Upon further consideration I've came to conclusion that partial copy/paste here is inevitable for the context of 1st commit. :( 

`datetime_to_tm` from `strftime.c` (2nd commit) is not using the standard `struct tm`, but rather local `struct dt_tm`, with slightly different set of fields, necessary for `strftime()` `"%z"` and `"%f"` flags. For context of 2nd commit though, we could change code to use global `datetime_to_tm()`  implementation, and call `tnt_strftime()` instead of `strftime()`. So this might end up with changes in both commits in this pull-request (though both will be maximally atomic)

Reply:
[But we may end up squashing 2nd commit to the 1st one, if these manipulations will not look much atomic]

Reply:
Well, I've changed things a bit:
- got rid of local copypasta of `datetime_to_tm`, `local_rd` and `local_dt` in the `test/unit/datetime.c`;
- defined `struct datetime_tm` (extension of standard `struct tm` used by `tnt_strftime()`), 
 `datetime_to_dt` and `tnt_strftime` in the `datetime.h` header;
- used `tnt_strftime` and `datetime_to_tm` in the `test/unit/datetime.c`;
- I had to modify format used from "%F %T" to "%F %T%z" because `datetime_to_tm()` now correctly
  process `tzoffset`, so output strings are now correct and timezone based, not GMT-based;

<details>
  <summary>Click to see patch!</summary>

```diff
diff --git a/src/lib/core/datetime.h b/src/lib/core/datetime.h
index 3d5f520e3..9d197a826 100644
--- a/src/lib/core/datetime.h
+++ b/src/lib/core/datetime.h
@@ -77,6 +77,57 @@ tnt_datetime_strftime(const struct datetime *date, char *buf, size_t len,
 void
 tnt_datetime_now(struct datetime *now);
 
+/**
+ * Local version resembling ISO C' tm structure.
+ * Includes original epoch value, and nanoseconds.
+ */
+struct datetime_tm {
+	/** Seconds. [0-60] (1 leap second) */
+	int tm_sec;
+	/** Minutes. [0-59] */
+	int tm_min;
+	/** Hours. [0-23] */
+	int tm_hour;
+	/** Day. [1-31] */
+	int tm_mday;
+	/** Month. [0-11] */
+	int tm_mon;
+	/** Year - 1900. */
+	int tm_year;
+	/** Day of week. [0-6] */
+	int tm_wday;
+	/** Days in year.[0-365] */
+	int tm_yday;
+	/** DST. [-1/0/1] */
+	int tm_isdst;
+
+	/** Seconds east of UTC. */
+	long int tm_gmtoff;
+	/** Timezone abbreviation. */
+	const char *tm_zone;
+	/** Seconds since Epoch */
+	int64_t tm_epoch;
+	/** nanoseconds */
+	int tm_nsec;
+};
+
+/**
+ * Convert @sa datetime to @sa dt_tm
+ */
+void
+datetime_to_tm(const struct datetime *date, struct datetime_tm *tm);
+
+/**
+ * tnt_strftime is Tarantol version of POSIX' strftime()
+ * which has been extended with %f (fractions of second)
+ * flag support. In all other aspect it's behaving exactly
+ * like standard strftime.
+ * @sa strftime()
+ */
+size_t
+tnt_strftime(char *s, size_t maxsize, const char *format,
+	     const struct datetime_tm *tm);
+
 #if defined(__cplusplus)
 } /* extern "C" */
 #endif /* defined(__cplusplus) */
diff --git a/test/unit/CMakeLists.txt b/test/unit/CMakeLists.txt
index 5662e89f2..9da06fe86 100644
--- a/test/unit/CMakeLists.txt
+++ b/test/unit/CMakeLists.txt
@@ -59,7 +59,7 @@ target_link_libraries(random.test core unit)
 add_executable(xmalloc.test xmalloc.c core_test_utils.c)
 target_link_libraries(xmalloc.test core unit)
 add_executable(datetime.test datetime.c)
-target_link_libraries(datetime.test cdt core unit)
+target_link_libraries(datetime.test tzcode cdt core unit)
 
 add_executable(bps_tree.test bps_tree.cc)
 target_link_libraries(bps_tree.test small misc)
diff --git a/test/unit/datetime.c b/test/unit/datetime.c
index d3b1d4222..4665ffef1 100644
--- a/test/unit/datetime.c
+++ b/test/unit/datetime.c
@@ -136,34 +136,6 @@ exit:
 	return 0;
 }
 
-static int
-local_rd(const struct datetime *dt)
-{
-	return (int)((int64_t)dt->epoch / SECS_PER_DAY) + DT_EPOCH_1970_OFFSET;
-}
-
-static int
-local_dt(const struct datetime *dt)
-{
-	return dt_from_rdn(local_rd(dt));
-}
-
-struct tm *
-datetime_to_tm(struct datetime *dt)
-{
-	static struct tm tm;
-
-	memset(&tm, 0, sizeof(tm));
-	dt_to_struct_tm(local_dt(dt), &tm);
-
-	int seconds_of_day = (int64_t)dt->epoch % 86400;
-	tm.tm_hour = (seconds_of_day / 3600) % 24;
-	tm.tm_min = (seconds_of_day / 60) % 60;
-	tm.tm_sec = seconds_of_day % 60;
-
-	return &tm;
-}
-
 static void
 datetime_test(void)
 {
@@ -193,17 +165,18 @@ datetime_test(void)
 		 */
 		static char buff[40];
 		struct datetime dt = { secs, nanosecs, offset, 0 };
+		struct datetime_tm tm = { .tm_epoch = 0 };
 		/* datetime_to_tm returns time in GMT zone */
-		struct tm *p_tm = datetime_to_tm(&dt);
-		size_t len = strftime(buff, sizeof(buff), "%F %T", p_tm);
+		datetime_to_tm(&dt, &tm);
+		size_t len = tnt_strftime(buff, sizeof(buff), "%F %T%z", &tm);
 		ok(len > 0, "strftime");
 		int64_t parsed_secs;
 		int32_t parsed_nsecs, parsed_ofs;
 		rc = parse_datetime(buff, len, &parsed_secs, &parsed_nsecs,
 				    &parsed_ofs);
 		is(rc, 0, "correct parse_datetime return value for '%s'", buff);
-		is(secs, parsed_secs, "reversible seconds via strftime for '%s",
-		   buff);
+		is(secs, parsed_secs,
+		   "reversible seconds via strftime for '%s'", buff);
 	}
 	check_plan();
 }
diff --git a/test/unit/datetime.result b/test/unit/datetime.result
index 19d07b25e..e9b07efa8 100644
--- a/test/unit/datetime.result
+++ b/test/unit/datetime.result
@@ -3,358 +3,358 @@
     ok 1 - correct parse_datetime return value for '2012-12-24 15:30Z'
     ok 2 - correct parse_datetime output seconds for '2012-12-24 15:30Z
     ok 3 - strftime
-    ok 4 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 5 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 4 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 5 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 6 - correct parse_datetime return value for '2012-12-24 15:30z'
     ok 7 - correct parse_datetime output seconds for '2012-12-24 15:30z
     ok 8 - strftime
-    ok 9 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 10 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 9 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 10 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 11 - correct parse_datetime return value for '2012-12-24 15:30'
     ok 12 - correct parse_datetime output seconds for '2012-12-24 15:30
     ok 13 - strftime
-    ok 14 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 15 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 14 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 15 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 16 - correct parse_datetime return value for '2012-12-24 16:30+01:00'
     ok 17 - correct parse_datetime output seconds for '2012-12-24 16:30+01:00
     ok 18 - strftime
-    ok 19 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 20 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 19 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 20 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 21 - correct parse_datetime return value for '2012-12-24 16:30+0100'
     ok 22 - correct parse_datetime output seconds for '2012-12-24 16:30+0100
     ok 23 - strftime
-    ok 24 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 25 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 24 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 25 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 26 - correct parse_datetime return value for '2012-12-24 16:30+01'
     ok 27 - correct parse_datetime output seconds for '2012-12-24 16:30+01
     ok 28 - strftime
-    ok 29 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 30 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 29 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 30 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 31 - correct parse_datetime return value for '2012-12-24 14:30-01:00'
     ok 32 - correct parse_datetime output seconds for '2012-12-24 14:30-01:00
     ok 33 - strftime
-    ok 34 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 35 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 34 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 35 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 36 - correct parse_datetime return value for '2012-12-24 14:30-0100'
     ok 37 - correct parse_datetime output seconds for '2012-12-24 14:30-0100
     ok 38 - strftime
-    ok 39 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 40 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 39 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 40 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 41 - correct parse_datetime return value for '2012-12-24 14:30-01'
     ok 42 - correct parse_datetime output seconds for '2012-12-24 14:30-01
     ok 43 - strftime
-    ok 44 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 45 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 44 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 45 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 46 - correct parse_datetime return value for '2012-12-24 15:30:00Z'
     ok 47 - correct parse_datetime output seconds for '2012-12-24 15:30:00Z
     ok 48 - strftime
-    ok 49 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 50 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 49 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 50 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 51 - correct parse_datetime return value for '2012-12-24 15:30:00z'
     ok 52 - correct parse_datetime output seconds for '2012-12-24 15:30:00z
     ok 53 - strftime
-    ok 54 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 55 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 54 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 55 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 56 - correct parse_datetime return value for '2012-12-24 15:30:00'
     ok 57 - correct parse_datetime output seconds for '2012-12-24 15:30:00
     ok 58 - strftime
-    ok 59 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 60 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 59 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 60 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 61 - correct parse_datetime return value for '2012-12-24 16:30:00+01:00'
     ok 62 - correct parse_datetime output seconds for '2012-12-24 16:30:00+01:00
     ok 63 - strftime
-    ok 64 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 65 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 64 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 65 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 66 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
     ok 67 - correct parse_datetime output seconds for '2012-12-24 16:30:00+0100
     ok 68 - strftime
-    ok 69 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 70 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 69 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 70 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 71 - correct parse_datetime return value for '2012-12-24 14:30:00-01:00'
     ok 72 - correct parse_datetime output seconds for '2012-12-24 14:30:00-01:00
     ok 73 - strftime
-    ok 74 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 75 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 74 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 75 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 76 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
     ok 77 - correct parse_datetime output seconds for '2012-12-24 14:30:00-0100
     ok 78 - strftime
-    ok 79 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 80 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 79 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 80 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 81 - correct parse_datetime return value for '2012-12-24 15:30:00.123456Z'
     ok 82 - correct parse_datetime output seconds for '2012-12-24 15:30:00.123456Z
     ok 83 - strftime
-    ok 84 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 85 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 84 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 85 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 86 - correct parse_datetime return value for '2012-12-24 15:30:00.123456z'
     ok 87 - correct parse_datetime output seconds for '2012-12-24 15:30:00.123456z
     ok 88 - strftime
-    ok 89 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 90 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 89 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 90 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 91 - correct parse_datetime return value for '2012-12-24 15:30:00.123456'
     ok 92 - correct parse_datetime output seconds for '2012-12-24 15:30:00.123456
     ok 93 - strftime
-    ok 94 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 95 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 94 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 95 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 96 - correct parse_datetime return value for '2012-12-24 16:30:00.123456+01:00'
     ok 97 - correct parse_datetime output seconds for '2012-12-24 16:30:00.123456+01:00
     ok 98 - strftime
-    ok 99 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 100 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 99 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 100 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 101 - correct parse_datetime return value for '2012-12-24 16:30:00.123456+01'
     ok 102 - correct parse_datetime output seconds for '2012-12-24 16:30:00.123456+01
     ok 103 - strftime
-    ok 104 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 105 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 104 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 105 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 106 - correct parse_datetime return value for '2012-12-24 14:30:00.123456-01:00'
     ok 107 - correct parse_datetime output seconds for '2012-12-24 14:30:00.123456-01:00
     ok 108 - strftime
-    ok 109 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 110 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 109 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 110 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 111 - correct parse_datetime return value for '2012-12-24 14:30:00.123456-01'
     ok 112 - correct parse_datetime output seconds for '2012-12-24 14:30:00.123456-01
     ok 113 - strftime
-    ok 114 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 115 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 114 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 115 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 116 - correct parse_datetime return value for '2012-12-24t15:30Z'
     ok 117 - correct parse_datetime output seconds for '2012-12-24t15:30Z
     ok 118 - strftime
-    ok 119 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 120 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 119 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 120 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 121 - correct parse_datetime return value for '2012-12-24t15:30z'
     ok 122 - correct parse_datetime output seconds for '2012-12-24t15:30z
     ok 123 - strftime
-    ok 124 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 125 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 124 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 125 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 126 - correct parse_datetime return value for '2012-12-24t15:30'
     ok 127 - correct parse_datetime output seconds for '2012-12-24t15:30
     ok 128 - strftime
-    ok 129 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 130 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 129 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 130 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 131 - correct parse_datetime return value for '2012-12-24t16:30+01:00'
     ok 132 - correct parse_datetime output seconds for '2012-12-24t16:30+01:00
     ok 133 - strftime
-    ok 134 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 135 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 134 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 135 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 136 - correct parse_datetime return value for '2012-12-24t16:30+0100'
     ok 137 - correct parse_datetime output seconds for '2012-12-24t16:30+0100
     ok 138 - strftime
-    ok 139 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 140 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 139 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 140 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 141 - correct parse_datetime return value for '2012-12-24t14:30-01:00'
     ok 142 - correct parse_datetime output seconds for '2012-12-24t14:30-01:00
     ok 143 - strftime
-    ok 144 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 145 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 144 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 145 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 146 - correct parse_datetime return value for '2012-12-24t14:30-0100'
     ok 147 - correct parse_datetime output seconds for '2012-12-24t14:30-0100
     ok 148 - strftime
-    ok 149 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 150 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 149 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 150 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 151 - correct parse_datetime return value for '2012-12-24t15:30:00Z'
     ok 152 - correct parse_datetime output seconds for '2012-12-24t15:30:00Z
     ok 153 - strftime
-    ok 154 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 155 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 154 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 155 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 156 - correct parse_datetime return value for '2012-12-24t15:30:00z'
     ok 157 - correct parse_datetime output seconds for '2012-12-24t15:30:00z
     ok 158 - strftime
-    ok 159 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 160 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 159 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 160 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 161 - correct parse_datetime return value for '2012-12-24t15:30:00'
     ok 162 - correct parse_datetime output seconds for '2012-12-24t15:30:00
     ok 163 - strftime
-    ok 164 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 165 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 164 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 165 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 166 - correct parse_datetime return value for '2012-12-24t16:30:00+01:00'
     ok 167 - correct parse_datetime output seconds for '2012-12-24t16:30:00+01:00
     ok 168 - strftime
-    ok 169 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 170 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 169 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 170 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 171 - correct parse_datetime return value for '2012-12-24t16:30:00+0100'
     ok 172 - correct parse_datetime output seconds for '2012-12-24t16:30:00+0100
     ok 173 - strftime
-    ok 174 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 175 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 174 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 175 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 176 - correct parse_datetime return value for '2012-12-24t14:30:00-01:00'
     ok 177 - correct parse_datetime output seconds for '2012-12-24t14:30:00-01:00
     ok 178 - strftime
-    ok 179 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 180 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 179 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 180 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 181 - correct parse_datetime return value for '2012-12-24t14:30:00-0100'
     ok 182 - correct parse_datetime output seconds for '2012-12-24t14:30:00-0100
     ok 183 - strftime
-    ok 184 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 185 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 184 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 185 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 186 - correct parse_datetime return value for '2012-12-24t15:30:00.123456Z'
     ok 187 - correct parse_datetime output seconds for '2012-12-24t15:30:00.123456Z
     ok 188 - strftime
-    ok 189 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 190 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 189 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 190 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 191 - correct parse_datetime return value for '2012-12-24t15:30:00.123456z'
     ok 192 - correct parse_datetime output seconds for '2012-12-24t15:30:00.123456z
     ok 193 - strftime
-    ok 194 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 195 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 194 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 195 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 196 - correct parse_datetime return value for '2012-12-24t16:30:00.123456+01:00'
     ok 197 - correct parse_datetime output seconds for '2012-12-24t16:30:00.123456+01:00
     ok 198 - strftime
-    ok 199 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 200 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 199 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 200 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 201 - correct parse_datetime return value for '2012-12-24t14:30:00.123456-01:00'
     ok 202 - correct parse_datetime output seconds for '2012-12-24t14:30:00.123456-01:00
     ok 203 - strftime
-    ok 204 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 205 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 204 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 205 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 206 - correct parse_datetime return value for '2012-12-24 16:30 +01:00'
     ok 207 - correct parse_datetime output seconds for '2012-12-24 16:30 +01:00
     ok 208 - strftime
-    ok 209 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 210 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 209 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 210 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 211 - correct parse_datetime return value for '2012-12-24 14:30 -01:00'
     ok 212 - correct parse_datetime output seconds for '2012-12-24 14:30 -01:00
     ok 213 - strftime
-    ok 214 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 215 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 214 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 215 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 216 - correct parse_datetime return value for '2012-12-24 15:30 UTC'
     ok 217 - correct parse_datetime output seconds for '2012-12-24 15:30 UTC
     ok 218 - strftime
-    ok 219 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 220 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 219 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 220 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 221 - correct parse_datetime return value for '2012-12-24 16:30 UTC+1'
     ok 222 - correct parse_datetime output seconds for '2012-12-24 16:30 UTC+1
     ok 223 - strftime
-    ok 224 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 225 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 224 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 225 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 226 - correct parse_datetime return value for '2012-12-24 16:30 UTC+01'
     ok 227 - correct parse_datetime output seconds for '2012-12-24 16:30 UTC+01
     ok 228 - strftime
-    ok 229 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 230 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 229 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 230 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 231 - correct parse_datetime return value for '2012-12-24 16:30 UTC+0100'
     ok 232 - correct parse_datetime output seconds for '2012-12-24 16:30 UTC+0100
     ok 233 - strftime
-    ok 234 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 235 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 234 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 235 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 236 - correct parse_datetime return value for '2012-12-24 16:30 UTC+01:00'
     ok 237 - correct parse_datetime output seconds for '2012-12-24 16:30 UTC+01:00
     ok 238 - strftime
-    ok 239 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 240 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 239 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 240 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 241 - correct parse_datetime return value for '2012-12-24 14:30 UTC-1'
     ok 242 - correct parse_datetime output seconds for '2012-12-24 14:30 UTC-1
     ok 243 - strftime
-    ok 244 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 245 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 244 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 245 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 246 - correct parse_datetime return value for '2012-12-24 14:30 UTC-01'
     ok 247 - correct parse_datetime output seconds for '2012-12-24 14:30 UTC-01
     ok 248 - strftime
-    ok 249 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 250 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 249 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 250 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 251 - correct parse_datetime return value for '2012-12-24 14:30 UTC-01:00'
     ok 252 - correct parse_datetime output seconds for '2012-12-24 14:30 UTC-01:00
     ok 253 - strftime
-    ok 254 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 255 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 254 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 255 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 256 - correct parse_datetime return value for '2012-12-24 14:30 UTC-0100'
     ok 257 - correct parse_datetime output seconds for '2012-12-24 14:30 UTC-0100
     ok 258 - strftime
-    ok 259 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 260 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 259 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 260 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 261 - correct parse_datetime return value for '2012-12-24 15:30 GMT'
     ok 262 - correct parse_datetime output seconds for '2012-12-24 15:30 GMT
     ok 263 - strftime
-    ok 264 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 265 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 264 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 265 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 266 - correct parse_datetime return value for '2012-12-24 16:30 GMT+1'
     ok 267 - correct parse_datetime output seconds for '2012-12-24 16:30 GMT+1
     ok 268 - strftime
-    ok 269 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 270 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 269 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 270 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 271 - correct parse_datetime return value for '2012-12-24 16:30 GMT+01'
     ok 272 - correct parse_datetime output seconds for '2012-12-24 16:30 GMT+01
     ok 273 - strftime
-    ok 274 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 275 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 274 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 275 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 276 - correct parse_datetime return value for '2012-12-24 16:30 GMT+0100'
     ok 277 - correct parse_datetime output seconds for '2012-12-24 16:30 GMT+0100
     ok 278 - strftime
-    ok 279 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 280 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 279 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 280 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 281 - correct parse_datetime return value for '2012-12-24 16:30 GMT+01:00'
     ok 282 - correct parse_datetime output seconds for '2012-12-24 16:30 GMT+01:00
     ok 283 - strftime
-    ok 284 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 285 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 284 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 285 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 286 - correct parse_datetime return value for '2012-12-24 14:30 GMT-1'
     ok 287 - correct parse_datetime output seconds for '2012-12-24 14:30 GMT-1
     ok 288 - strftime
-    ok 289 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 290 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 289 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 290 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 291 - correct parse_datetime return value for '2012-12-24 14:30 GMT-01'
     ok 292 - correct parse_datetime output seconds for '2012-12-24 14:30 GMT-01
     ok 293 - strftime
-    ok 294 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 295 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 294 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 295 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 296 - correct parse_datetime return value for '2012-12-24 14:30 GMT-01:00'
     ok 297 - correct parse_datetime output seconds for '2012-12-24 14:30 GMT-01:00
     ok 298 - strftime
-    ok 299 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 300 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 299 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 300 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 301 - correct parse_datetime return value for '2012-12-24 14:30 GMT-0100'
     ok 302 - correct parse_datetime output seconds for '2012-12-24 14:30 GMT-0100
     ok 303 - strftime
-    ok 304 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 305 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 304 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 305 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 306 - correct parse_datetime return value for '2012-12-24 14:30 -01:00'
     ok 307 - correct parse_datetime output seconds for '2012-12-24 14:30 -01:00
     ok 308 - strftime
-    ok 309 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 310 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 309 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 310 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 311 - correct parse_datetime return value for '2012-12-24 16:30:00 +01:00'
     ok 312 - correct parse_datetime output seconds for '2012-12-24 16:30:00 +01:00
     ok 313 - strftime
-    ok 314 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 315 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 314 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 315 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 316 - correct parse_datetime return value for '2012-12-24 14:30:00 -01:00'
     ok 317 - correct parse_datetime output seconds for '2012-12-24 14:30:00 -01:00
     ok 318 - strftime
-    ok 319 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 320 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 319 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 320 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 321 - correct parse_datetime return value for '2012-12-24 16:30:00.123456 +01:00'
     ok 322 - correct parse_datetime output seconds for '2012-12-24 16:30:00.123456 +01:00
     ok 323 - strftime
-    ok 324 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 325 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 324 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 325 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 326 - correct parse_datetime return value for '2012-12-24 14:30:00.123456 -01:00'
     ok 327 - correct parse_datetime output seconds for '2012-12-24 14:30:00.123456 -01:00
     ok 328 - strftime
-    ok 329 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 330 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 329 - correct parse_datetime return value for '2012-12-24 14:30:00-0100'
+    ok 330 - reversible seconds via strftime for '2012-12-24 14:30:00-0100'
     ok 331 - correct parse_datetime return value for '2012-12-24 15:30:00.123456 -00:00'
     ok 332 - correct parse_datetime output seconds for '2012-12-24 15:30:00.123456 -00:00
     ok 333 - strftime
-    ok 334 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 335 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 334 - correct parse_datetime return value for '2012-12-24 15:30:00+0000'
+    ok 335 - reversible seconds via strftime for '2012-12-24 15:30:00+0000'
     ok 336 - correct parse_datetime return value for '20121224T1630+01:00'
     ok 337 - correct parse_datetime output seconds for '20121224T1630+01:00
     ok 338 - strftime
-    ok 339 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 340 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 339 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 340 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 341 - correct parse_datetime return value for '2012-12-24T1630+01:00'
     ok 342 - correct parse_datetime output seconds for '2012-12-24T1630+01:00
     ok 343 - strftime
-    ok 344 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 345 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 344 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 345 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 346 - correct parse_datetime return value for '20121224T16:30+01'
     ok 347 - correct parse_datetime output seconds for '20121224T16:30+01
     ok 348 - strftime
-    ok 349 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 350 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 349 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 350 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
     ok 351 - correct parse_datetime return value for '20121224T16:30 +01'
     ok 352 - correct parse_datetime output seconds for '20121224T16:30 +01
     ok 353 - strftime
-    ok 354 - correct parse_datetime return value for '2012-12-24 15:30:00'
-    ok 355 - reversible seconds via strftime for '2012-12-24 15:30:00
+    ok 354 - correct parse_datetime return value for '2012-12-24 16:30:00+0100'
+    ok 355 - reversible seconds via strftime for '2012-12-24 16:30:00+0100'
 ok 1 - subtests
     1..15
     ok 1 - string '1970-01-01T02:00:00+0200' expected, received '1970-01-01T02:00:00+0200'
diff --git a/third_party/tzcode/strftime.c b/third_party/tzcode/strftime.c
index db19495c5..78dc81064 100644
--- a/third_party/tzcode/strftime.c
+++ b/third_party/tzcode/strftime.c
@@ -108,30 +108,12 @@ enum warn {
 	IN_ALL
 };
 
-/* ISO C `broken-down time' structure.  */
-struct dt_tm {
-	int tm_sec;		/* Seconds.     [0-60] (1 leap second) */
-	int tm_min;		/* Minutes.     [0-59] */
-	int tm_hour;		/* Hours.       [0-23] */
-	int tm_mday;		/* Day.         [1-31] */
-	int tm_mon;		/* Month.       [0-11] */
-	int tm_year;		/* Year - 1900.  */
-	int tm_wday;		/* Day of week. [0-6] */
-	int tm_yday;		/* Days in year.[0-365] */
-	int tm_isdst;		/* DST.         [-1/0/1] */
-
-	long int tm_gmtoff;	/* Seconds east of UTC.  */
-	const char *tm_zone;	/* Timezone abbreviation.  */
-	int64_t tm_epoch;	/* Seconds since Epoch */
-	int tm_nsec;		/* nanoseconds  */
-};
-
 static char *
 _add(const char *, char *, const char *);
 static char *
 _conv(int, const char *, char *, const char *);
 static char *
-_fmt(const char *, const struct dt_tm *, char *, const char *, enum warn *);
+_fmt(const char *, const struct datetime_tm *, char *, const char *, enum warn *);
 static char *
 _yconv(int, int, bool, bool, char *, char const *);
 
@@ -153,7 +135,7 @@ local_secs(const struct datetime *date)
 }
 
 void
-datetime_to_tm(const struct datetime *date, struct dt_tm *tm)
+datetime_to_tm(const struct datetime *date, struct datetime_tm *tm)
 {
 	struct tm t;
 	memset(&t, 0, sizeof(t));
@@ -174,8 +156,9 @@ datetime_to_tm(const struct datetime *date, struct dt_tm *tm)
 	tm->tm_sec = seconds_of_day % 60;
 }
 
-static size_t
-tnt_strftime(char *s, size_t maxsize, const char *format, const struct dt_tm *t)
+size_t
+tnt_strftime(char *s, size_t maxsize, const char *format,
+	     const struct datetime_tm *t)
 {
 	char *p;
 	int saved_errno = errno;
@@ -200,13 +183,13 @@ size_t
 tnt_datetime_strftime(const struct datetime *date, char *buf, size_t len,
 		      const char *fmt)
 {
-	struct dt_tm tm;
+	struct datetime_tm tm;
 	datetime_to_tm(date, &tm);
 	return tnt_strftime(buf, len, fmt, &tm);
 }
 
 static char *
-_fmt(const char *format, const struct dt_tm *t, char *pt, const char *ptlim,
+_fmt(const char *format, const struct datetime_tm *t, char *pt, const char *ptlim,
      enum warn *warnp)
 {
 	for (; *format; ++format) {
```
</details>